### PR TITLE
Map time data 1

### PR DESCRIPTION
**Pull request by geramirez**
I decided to split 18F/foia/pull/81 into two pieces. Part one re-works some of the details of the mapping from foia.gov/data to foia hub. Below are examples of some of the mapping issues I encountered. In all, about 160 of the names between foia.gov/data and foia_hub don't match. Much of this mapping was completed manually in order to ensure accuracy. 

```
**spelling errors**
National Cemetary Administration ---> National Cemetery Administration  
Bureau of Alcohol, Tobacco, Firearms, and Explosives ---> Bureau of Alcohal, Tobacco, Firearms and Explosives

**& vs and both ways**
Office for Civil Rights and Civil Liberties ---> Office of Civil Rights & Civil Liberties
Veterans' Employment & Training Service ---> Veterans' Employment and Training Service

**multiple mappings**
Savannah River Office --> Savannah River Office AND Savannah River Operations Office
Office of Information and Technology --->  Office of Information & Technology AND Office of Information and Technology (005) AND Office of Information and Technology

**additions**
Office of the Assistant Secretary for Operations, Security, and Preparedness ---> Office of Operations, Security, and Preparedness (formerly S&LE - Security & Law Enforcement)
Office of Special Counsel ---> U.S. Office of Special Counsel

**omissions and number changes**
Region V (Chicago, Columbus, Cleveland, Detroit, Indianapolis, Milwaukee, Minneapolis-St. Paul) ---> Region 5 - Chicago

```

### CODE REVIEW

File: contacts/processing_time_scraper.py [hunk 1]
```diff
@@ -44,24 +45,36 @@ def delete_empty_data(data):
...
+        if data[key] == '':
             del data[key]
     return data
 
 
-def append_time_stats(yaml_data, data, year, short_filename):
+def clean_data(data):
+    """
+    Deletes agency, year, and component attributes, which are not
+    added to the yamls and also any attributs with empty values
```
Comment (khandelwal) [inline contacts/processing_time_scraper.py]: s/attributs/attributes

File: contacts/tests/tests.py [hunk 2]
```diff
@@ -704,3 +704,21 @@ def test_clean_html(self):
...
+        """ Should delete any items with a value of '' """
+
+        test_data = {'A': '', 'B': 'value B'}
+        returned_data = processing_time_scraper.delete_empty_data(test_data)
+        self.assertEqual(returned_data, {'B': 'value B'})
+
+    def test_clean_data(self):
+        """
+        Should deletes agency, year, and component attributes, which are not
+        added to the yamls and also any attributs with empty values
```
Comment (khandelwal) [inline contacts/tests/tests.py]: attributes

File: contacts/processing_time_scraper.py [hunk 3]
```diff
@@ -31,11 +32,11 @@ def apply_mapping(data):
...
     mapping = load_mapping()
     new_data = {}
-    for key in data.keys():
-        if key in mapping.keys():
-            new_data[mapping[key]] = data[key]
-        else:
-            new_data[key] = data[key]
+    for yaml_key in mapping.keys():
+        if mapping[yaml_key] in data.keys():
+            new_data[yaml_key] = deepcopy(data[mapping[yaml_key]])
```
Comment (khandelwal) [inline contacts/processing_time_scraper.py]: Do we really need to deepcopy here?

**Comment by khandelwal** [comment]
I tried to view the mapping data file but saw this error (from Github). Is there something weird going on with row 14? 

"We can make this file beautiful and searchable if this error is corrected: It looks like row 14 should actually have 4 columns, instead of 5."

Also, are there commas missing in the file? When I view the 'raw' version there certainly seem to be. 

Also, we've sort of been using YAML to store data. Would this file be much easier to read if it was in YAML?

**Comment by khandelwal** [comment]
Also, you're doing some name mapping related stuff - but I didn't see any tests around that?

**Comment by geramirez** [comment]
@khandelwal Okay, I switched to yamls and added tests. I'm going to leave the PR assigned to me until I double-check that all the data that was removed makes sense and create a file with all the data from foia.gov that wasn't matched.

I'm sure there are a couple more elements that we could map but I took a conservative approach. The ones left might require a bit more research.

**Comment by geramirez** [comment]
[Unmatched records](https://gist.github.com/ramirezg/3d7c43aadeb3741b1ba2) - I noticed that these records contain many regional offices. I wonder if they might be able to help validate agency sub-components from 18F/foia-hub/issues/251

**Comment by khandelwal** [comment]
I'm really confused with the mapping file. Here's an entry: 

Bureau of Engraving and PrintingDepartment of the Treasury:
  foiacontacts:
- Bureau of Engraving and PrintingDepartment of the Treasury
  foiadata:
- bureau of engraving and printing_treasury
  foiayamls:
- Bureau of Engraving and Printing_Treasury
1. What is foiadata, foiacontacts and foiayamls?
2. Why does the foiacontacts entry not have a space between the two names? 
3. In almost all cases foiadata is a lower case foia_yamls (do we really need an explicit mapping for this?)

**Comment by geramirez** [comment]
1.
foiadata = naming convention from foia.gov/data formatted (name_top-level)
foiayamls = naming conventions from the yaml files formatted (name_filename)
foiacontacts = naming convention from the foia contacts xls formatted (officename+agencyname)

2.
It's just a concatenation I can change it.

3.
No, I can take them out, but it's easier to add or make changes to the mapping if everything is explicit. I can also change foiayamls to all lowercase.

**Comment by khandelwal** [comment]
Why is the yamls name for things different than the foiadata name for things? Doesn't the name of the agency in our dataset come from foia.gov/data? 

The concatenation doesn't make sense. Remove it or use a separating character.

**Comment by khandelwal** [comment]
If there's no difference between the two names after the lowercase, I see no reason to include that mapping. It's just confusing at that point.

**Comment by geramirez** [comment]
@khandelwal The names from foia.gov/data are different than the names from our yaml files because foia.gov/data has a completely different naming conventions than foia.gov/report-makerequest, which is were we first scraped our data. At first I assumed that foia.gov was internally consistent, but it's actually not. 

I'll take care of the separating characters and removing the similar strings.

**Comment by khandelwal** [comment]
Ah, that makes more sense. Do YAML files support comments? If that's the case - let's add that fact as a comment, so that it's very clear what the mapping file is.

**Comment by geramirez** [comment]
I made the changes. I left out the foia contacts mapping out for another PR.

**Pull request merged**